package.json

```diff
@@ -78,10 +80,21 @@
     "webpack-merge": "^4.1.5",
     "webpack-node-externals": "^1.7.2"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.{js,vue}": [
+      "npm run lint:autofix",
+      "git add"
```

Comment:
will this add all the files automatically?

Reply:
Not all. the staged file only.

Reply:
I see. What if don't want the full file to be staged? It happens rarely, but sometimes I like to `git add -p` and only add parts of a file. Will this be impossible?

I appreciate adding the commit hook but I think we should just lint on CI.

Reply:
I tried with the `git add -p`. It does not touch whatever you wish not to commit. So technically this only run on the staged item. My earlier explanation is wrong, not staged file but staged item(can be file/line). 

With this, we can catch the lint error earlier, does not need to wait for CI build. My self a bit annoyed when the build failed just because of lint but we can remove this if lint on CI is good enough.